portfolio/src/main/java/com/google/sps/servlets/ScoreServlet.java

```diff
@@ -0,0 +1,59 @@
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.ArrayList;
+
+import com.google.sps.data.Score;
+
+@WebServlet("/scores")
+public class ScoreServlet extends HttpServlet {
+    private DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException{
+        
+        // Get score limit parameter.
+        int max = Integer.parseInt(request.getParameter("max"));
```

Comment:
Maybe put this statement into the try / catch block.

Reply:
Added try catch block and client side alert for "Invalid page number."